pkg/ottl/paths.go

```diff
@@ -30,3 +34,53 @@ func getBooleanExpressionPaths(be *booleanExpression) []path {
 	be.accept(visitor)
 	return visitor.paths
 }
+
+// AppendStatementPathsContext changes the given OTTL statement adding the context name prefix
+// to all context-less paths. No modifications are performed for paths which [Path.Context]
+// value matches any WithPathContextNames value.
+// The context argument must be valid WithPathContextNames value, otherwise an error is returned.
+func (p *Parser[K]) AppendStatementPathsContext(context string, statement string) (string, error) {
+	if _, ok := p.pathContextNames[context]; !ok {
+		return statement, fmt.Errorf(`unknown context "%s" for parser %T, valid options are: %s`, context, p, p.buildPathContextNamesText(""))
+	}
+	parsed, err := parseStatement(statement)
+	if err != nil {
+		return "", err
+	}
+	paths := getParsedStatementPaths(parsed)
+	if len(paths) == 0 {
+		return statement, nil
+	}
+
+	var missingContextOffsets []int
+	for _, it := range paths {
+		if _, ok := p.pathContextNames[it.Context]; !ok {
+			missingContextOffsets = append(missingContextOffsets, it.Pos.Offset)
+		}
+	}
+
+	return writeStatementWithPathsContext(context, statement, missingContextOffsets), nil
```

Comment:
Would it be helpful or annoying for users if we wrote out an info log for the statements we changed? I want users to be aware of the changes we're making to their configuration. It would only be during startup, but could result in a log per statement, which could be a lot.

Reply:
The draft implementation is grouping and logging it in the [ParserCollection.logRewrittenStatements](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35050/files#diff-7efe3affb096a0fa16edab4c8e48d665bd120c938103b778ad778b15c5f04ce6R232), it emits a single message with all modified statements. I'd avoid logging it here for the same reason you mentioned.